File: SWEA/s1244.cpp
#include <iostream>
#include <algorithm>    // swap()
using namespace std;

int ch, res;
string num;
void dfs(int idx, int cur) {
    if(cur == ch) { // 교환 횟수 도달
        res = max(res, stoi(num));
        return;
    }

    for(int i=idx; i<num.size()-1; i++) {
        for(int j=i+1; j<num.size(); j++) {
            swap(num[i], num[j]);
            dfs(i, cur+1);
            swap(num[i], num[j]);
        }
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t;
    cin >> t;

    for(int i=1; i<=t; i++) {
        res = 0;
        cin >> num >> ch;

        if(ch > num.size()) {
            ch = num.size() - 1;
        }

        dfs(0, 0);

        cout << "#" << i << " " << res << '\n';
    }

    return 0;
}

File: SWEA/s1244.java
import java.util.*;

public class s1244 {
    static int ch, res;
    static char[] arr;
    static String str;

    public static void main(String[] args) throws Exception {

        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        for(int i=1; i<=t; i++) {
            res = 0;
            str = sc.next();
            arr = str.toCharArray();
            
            ch = sc.nextInt();

            dfs(0, 0);

            System.out.println("#" + i + " " + res);
        }

    }

    private static void dfs(int n, int idx) {
        if(n == ch) {
            res = Math.max(res, Integer.parseInt(str));
            return;
        }

        for(int i=idx; i<str.length(); i++) {
            for(int j=i+1; j<str.length(); j++) {
                if(str.charAt(i) <= str.charAt(j)) {
                    swap(i, j);
                    dfs(n+1, i);
                    swap(j, i);
                }
            }
        }
    }

    private static void swap(int i, int j) {
        StringBuilder chNum = new StringBuilder(str);

        char A = str.charAt(i);
        char B = str.charAt(j);

        chNum.setCharAt(i, B);
        chNum.setCharAt(j, A);
        str = chNum.toString();
    }
}
